bot/cogs/moderation/silence.py

```diff
@@ -57,23 +70,30 @@ async def _notifier(self) -> None:
 class Silence(commands.Cog):
     """Commands for stopping channel messages for `verified` role in a channel."""
 
+    # Maps muted channel IDs to their previous overwrites for send_message and add_reactions.
+    # Overwrites are stored as JSON.
+    previous_overwrites = RedisCache()
+
+    # Maps muted channel IDs to POSIX timestamps of when they'll be unsilenced.
+    # A timestamp equal to -1 means it's indefinite.
+    unsilence_timestamps = RedisCache()
```

Comment:
Have you considered using just a single cache?

Since we're already serializing the perms as JSON, it could be something like:
```
{
  permissions: { ... },
  timestamp: 123
}
```
For the sake of "safety" I'd probably feel tempted to write a class `Silence`, instances of which can serialize themselves into such a JSON, and can be created from one. That would create more of a "protocol" for what is allowed to exist in the cache, but I realize that I'm basically reinventing a table here with permissions and timestamps being columns. It'd also make it so that if we wanted to add more data, we'd have to "migrate" existing cache entries. Probably an awful idea, but it makes me feel a little uneasy how the caches are written to from different methods and there is no guarantee that a key exists in both.

Reply:
Since the POSIX timestamp comes out as a float, maybe we could write a `-1.` for consistent types, but eh nitpick.

Reply:
Keeping them separate makes it easier to separate permissions code from scheduling code. The caches are not intended to be used publicly so I'm not keen on overcomplicating serialisation for some extra safety. I could add some asserts if you'd like.

Reply:
It's fine, I agree that keeping it simple probably makes more sense.